lib/Alcaeus/MongoDbAdapter/AbstractCursor.php

```diff
@@ -127,15 +133,22 @@ public function key()
     }
 
     /**
-     * Advances the cursor to the next result
+     * Advances the cursor to the next result, and returns that result
      * @link http://www.php.net/manual/en/mongocursor.next.php
      * @throws \MongoConnectionException
      * @throws \MongoCursorTimeoutException
-     * @return void
+     * @return array Returns the next object
```

Comment:
The [Iterator::next() method](https://secure.php.net/manual/en/iterator.next.php) specifies void as return type. Is there a reason you're returning the current value after advancing the cursor?